api/v1alpha6/openstackcluster_types.go

```diff
@@ -112,6 +112,13 @@ type OpenStackClusterSpec struct {
 	// +optional
 	AllowAllInClusterTraffic bool `json:"allowAllInClusterTraffic"`
 
+	// SecurityGroupForCNI is used when you deploy different CNI
+	// AllowAll means allow all traffic
+	// Calico creates security group rules for Calico CNI and it's the default
+	// not supported value will print a warning and no security group applied
+	// +optional
+	SecurityGroupForCNI string `json:"securityGroupForCNI"`
```

Comment:
Here we can set the name of the CNI to use, e.g. calico?
So actually the value represents the CNI selection to decide, which security group rules to apply.
I'd propose to name it "CNIForSecurityGroup" or simply "CNI".

What does "AllowAll" mean in this description - the description could be misleading, as "AllowAll" looks like it is a valid value.
```suggestion
	// CNIForSecurityGroup defines the CNI to set the appropriate security group rules.
	// The default CNI used is "calico" and creates security group rules for Calico CNI.
	// Unsupported values create warnings and no security groups are applied.
	// +optional
	CNIForSecurityGroup string `json:"cniForSecurityGroup"`
```